hyperspy/_signals/eels.py

```diff
@@ -919,10 +1005,29 @@ def power_law_extrapolation(self,
             factor = s.axes_manager[-1].scale
         else:
             factor = 1
-        s.data[..., axis.size:] = (
-            factor * pl.A.map['values'][..., np.newaxis] *
-            s.axes_manager.signal_axes[0].axis[np.newaxis, axis.size:] ** (
-                -pl.r.map['values'][..., np.newaxis]))
+        if self.metadata.Signal.lazy:
```

Comment:
Why is lazy in metadata? It's not a property of the signal, is a way to deal with it.